File: src/main/java/im/tox/tox4j/core/callbacks/ToxEventListener.java
package im.tox.tox4j.core.callbacks;

public interface ToxEventListener extends
    SelfConnectionStatusCallback,
    FileRecvControlCallback,
    FileRecvCallback,
    FileRecvChunkCallback,
    FileChunkRequestCallback,
    FriendConnectionStatusCallback,
    FriendMessageCallback,
    FriendNameCallback,
    FriendRequestCallback,
    FriendStatusCallback,
    FriendStatusMessageCallback,
    FriendTypingCallback,
    FriendLosslessPacketCallback,
    FriendLossyPacketCallback,
    FriendReadReceiptCallback {

  ToxEventListener IGNORE = new ToxEventAdapter();
}

File: projects/tox4j/src/main/java/im/tox/tox4j/core/callbacks/ToxEventListener.scala
package im.tox.tox4j.core.callbacks

trait ToxEventListener extends
  SelfConnectionStatusCallback with
  FileRecvControlCallback with
  FileRecvCallback with
  FileRecvChunkCallback with
  FileChunkRequestCallback with
  FriendConnectionStatusCallback with
  FriendMessageCallback with
  FriendNameCallback with
  FriendRequestCallback with
  FriendStatusCallback with
  FriendStatusMessageCallback with
  FriendTypingCallback with
  FriendLosslessPacketCallback with
  FriendLossyPacketCallback with
  FriendReadReceiptCallback
